### ISSUE TITLE
Gracefully handle pykdtree's use of OpenMP (OMP_NUM_THREADS) with dask

### ISSUE THREAD

**Issue by djhoese** [pytroll/satpy#684]
Right now it is recommended in SatPy that the user set `OMP_NUM_THREADS` to limit the number of threads that each dask work uses when using the BLAS library (https://satpy.readthedocs.io/en/latest/faq.html). Without doing this you end up with `num_dask_workers * num_logical_cores` threads being created which can completely kill the kernel scheduler.

The problem is when we use libraries that don't operate on chunks of data and also use OpenMP, like `pykdtree`. In SatPy we have to create the KDTree object as one single chunk (longitude and latitude arrays as input). In this situation we'd like to give pykdtree as many OpenMP threads as it wants because it isn't benefiting from the parallelism of dask, but the environment variable currently suggested to users is only interpreted at load time (`OMP_NUM_THREADS`). I have seem some evidence from Geo2Grid users that this behavior is slowing down resampling way too much.

Here are some options I see:

1. Create a single dask multiprocess worker with OMP_NUM_THREADS unset or set to a keyword argument from the user. I have no idea if this is possible without getting extremely complicated. Theoretically this should be possible with `with dask.config.set(scheduler=new_single_process_scheduler):`. However, I think this has to be done during execution/compute which in the non-caching case is not possible.
2. [This numpy issue](https://github.com/numpy/numpy/issues/11826) discusses methods for setting OpenMP's number of threads dynamically at run time. It ends (at the time of writing) with the `threadpoolctl` package being created/advertised (https://github.com/joblib/threadpoolctl). I think this is our best long term solution until we get a non-KDTree based (or GPU based) nearest neighbor algorithm.

Any other ideas?

**Comment by djhoese** [comment]
I'm realizing now (based on pykdtree's benchmarks) that pykdtree probably doesn't use OpenMP for KDTree construction so this may only matter for querying...maybe.

**Comment by djhoese** [comment]
@pnuu what has your experience been with this. Maybe this isn't an actual issue. From what I can tell after looking at pykdtree's source code is that it doesn't use openmp for any of the construction but uses it for querying. Our querying *should* be per dask chunk so we should still be benefiting from dask's parallelism...right?

**Comment by pnuu** [comment]
I've never used pykdtree alone, only via SatPy and pyresample. And actually have never thought about how it affects the performance apart from the old timing comparisons against scipy's implementation.

**Comment by djhoese** [comment]
I'm going through old Polar2Grid issues and discovered the one linked above. I think I'm going to close this issue. The related numpy issue has been closed and suggests using threadpoolctl. We could use that but as mentioned above I realized that pykdtree doesn't use openmp for creation of the KDTree object so controlling this by allowing more openmp threads during KDTree creation wouldn't do anything.

We now have other techniques being developed in pyresample (better dask-friendly algorithms) that make this even more unnecessary.
